### PR TITLE
Revamp CI

### PR DESCRIPTION
**Pull request by ArnoStrouwen**

### CODE REVIEW

**Comment by ArnoStrouwen** [comment]
The remaining failures are stochastic.
I will not look into them further here.

**Comment by ChrisRackauckas** [comment]
This needs a rebase.

**Comment by ArnoStrouwen** [comment]
This is good to go, remaining failures are stochastic, and will not be fixed here:
https://github.com/SciML/HighDimPDE.jl/actions/runs/7595539675/job/20688222954#step:6:743
https://github.com/SciML/HighDimPDE.jl/actions/runs/7595539676/job/20688222881#step:7:2001

**Comment by ChrisRackauckas** [comment]
Can you just boost that bound to 0.17?

**Comment by ArnoStrouwen** [comment]
Now it returned 0.18 🤷.

File: test/DeepSplitting.jl [hunk 1]
```diff
@@ -107,7 +107,7 @@ end
         u1_anal = [u_anal(x, tspan[end]) for x in eachcol(xs)]
         e_l2 = mean(rel_error_l2.(u1, u1_anal))
         println("rel_error_l2 = ", e_l2, "\n")
-        @test e_l2 < 0.13
+        @test e_l2 < 0.17
```
Comment (ChrisRackauckas) [inline test/DeepSplitting.jl:110]:
```suggestion
        @test e_l2 < 0.185
```

**Pull request merged**